uint8 level
---
bool was_set
